mypy/util.py

```diff
@@ -313,3 +320,120 @@ def check_python_version(program: str) -> None:
     if sys.version_info[:3] == (3, 5, 0):
         sys.exit("Running {name} with Python 3.5.0 is not supported; "
                  "please upgrade to 3.5.1 or newer".format(name=program))
+
+
+def count_stats(errors: List[str]) -> Tuple[int, int]:
+    """Count total number of errors and files in error list."""
+    errors = [e for e in errors if ': error:' in e]
+    files = {e.split(':')[0] for e in errors}
+    return len(errors), len(files)
+
+
+class FancyFormatter:
+    """Apply color and bold font to terminal output.
+
+    This currently only works on Linus an Mac.
+    """
+    def __init__(self, f_out: IO[str], f_err: IO[str],
+                 show_error_codes: bool) -> None:
+        self.show_error_codes = show_error_codes
+        # Check if we are in a human-facing terminal on a supported platform.
+        if sys.platform not in ('linux', 'darwin'):
+            self.dummy_term = True
+            return
+        if not f_out.isatty() or not f_err.isatty():
```

Comment:
I think we need a flag to suppress the isatty checking, since our internal scripts all wrap mypy for logging purposes.

Reply:
Yes, indeed. I however added just an environment variable, it is probably relatively rarely needed thing.